fs-core/main/default/lwc/formulaShareRulesListView/formulaShareRulesListView.js

```diff
@@ -187,49 +186,60 @@ export default class TreeGrid extends NavigationMixin(LightningElement) {
         }
     }
 
+    namespacePrefix;
+    @wire(getNamespacePrefix)
+    wiredNamespacePrefix({ error, data })
+    {
+        if (data) {
+            console.log('>>>data: ' + JSON.stringify(data, null, '\t'));
+            this.namespacePrefix = data;
+        } else if (error) {
+            console.error('>>>error: ' + error);
+            this.namespacePrefix = undefined;
+            this.showError(error, 'Error getting namespace prefix.');
+        }
+    }
 
     // Subcribes to list platform event, and refresh treegrid each time event is received
     createOrUpdate = false;
     manageRefreshEvents() {
+        try {
+            //console.log('>>>this.namespacePrefix: '+ JSON.stringify(this.namespacePrefix, null, '\t'));
 
-        // Get namespace prefix
-        getNamespacePrefix()
-            .then((prefix) => {
-                //console.log('Got namespace: '+prefix);
+            // Subscribe to list update events (raised by batch job and on rule activate/deactivate)
+            const listUpdateCallback = (response) => {
+                //console.log('Received Refresh Event');
+                this.refreshView();
+            };
 
-                // Subscribe to list update events (raised by batch job and on rule activate/deactivate)
-                const listUpdateCallback = (response) => {
-                    //console.log('Received Refresh Event');
-                    this.refreshView();
-                };
-                subscribe('/event/'+prefix+'FormulaShare_List_Update__e', -1, listUpdateCallback).then(response => {
+            subscribe('/event/'+this.namespacePrefix+'FormulaShare_List_Update__e', -1, listUpdateCallback)
```

Comment:
I'm getting an issue testing this from the front end - when I click the "Submit for Recalculation" option for an object, the view won't be refreshed. Adding some debug statements I can see this happens because this.namespacePrefix is undefined when the subscribe call is made so the subscribe() function will be called with the parameter "/event/undefinedFormulaShare_List_Update__e".

The issue here I think is the component execution sequence - manageRefreshEvents() runs before the wire function has a chance to populate from apex.

The only way I know to get around this would be to revert getNamespacePrefix to run as an imperitive function - does that sound OK or is there a reason you'd changed this to a wire function?

Reply:
I've changed it back.